### PR TITLE
Test searching for puppet modules

### PR DESCRIPTION
**Pull request by Ichimonji10**
Test whether it is possible to search for and find puppet modules. In
particular, try to find puppet modules that belong to a certain repository or
content view version. Test results:

    $ nosetests tests/foreman/api/test_puppetmodule.py
    ..SS
    ----------------------------------------------------------------------
    Ran 4 tests in 75.308s

    OK (SKIP=2)

### CODE REVIEW

**Comment by Ichimonji10** [comment]
Depends on SatelliteQE/nailgun#200.

File: tests/foreman/api/test_puppetmodule.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        The repository belongs to ``cls.product``.
+
+        """
+        self.repository = entities.Repository(
+            content_type='puppet',
+            product=self.product,
+        ).create()
+
+    @skip_if_bug_open('bugzilla', 1260206)
+    def test_0(self):
```
Comment (elyezer) [inline tests/foreman/api/test_puppetmodule.py]: Can I haz better naming?

@oshtaier is on an effort to give meaningful names for all of our tests. Let's help him achieve it :smile:

**Comment by elyezer** [comment]
ACK pending comment.

File: tests/foreman/api/test_puppetmodule.py (hunk 1 above)
Comment (Ichimonji10) [inline tests/foreman/api/test_puppetmodule.py]: Sure. This can definitely be improved.

Comment (Ichimonji10) [inline tests/foreman/api/test_puppetmodule.py]: Done. I've renamed the methods like so:

* `test_0` → `test_0_search_results`
* `test_1` → `test_1_search_result`

**Comment by oshtaier** [comment]
ACK

**Pull request merged**